ens/src/eth-registrar.ts

```diff
@@ -0,0 +1,162 @@
+  let paymentToken: string | null = null
+  if (event.transaction.value > BIG_INT_ZERO) {
+    paymentToken = ZERO_ADDRESS;
+  }
+  airstack.domain.trackNameRegisteredTransaction(
+    event.transaction.hash.toHexString(),
+    event.block.number,
+    event.block.hash.toHexString(),
+    event.block.timestamp,
+    event.logIndex,
+    ETHEREUM_MAINNET_ID,
+    event.params.owner.toHexString(),
+    event.params.expires,
+    event.transaction.value,
+    paymentToken,
+    event.params.id,
+    rootNode,
+    TOKEN_ADDRESS_ENS,
+  );
+}
+
+/**
+ * @dev this function maps the NameRenewed event from the BaseRegistrar contract
+ * @param event NameRenewedEvent from BaseRegistrar contract
+ */
+export function handleNameRenewed(event: NameRenewedEvent): void {
+  log.info("handleNameRenewed: renewer {} label {} expiry {} txhash {}", [event.transaction.from.toHexString(), event.params.id.toHexString(), event.params.expires.toString(), event.transaction.hash.toHexString()]);
+  let paymentToken: string | null = null
+  if (event.transaction.value > BIG_INT_ZERO) {
+    paymentToken = ZERO_ADDRESS;
+  }
+  airstack.domain.trackNameRenewedTransaction(
+    event.transaction.hash.toHexString(),
+    event.block.number,
+    event.block.hash.toHexString(),
+    event.block.timestamp,
+    ETHEREUM_MAINNET_ID,
+    null,
+    paymentToken,
+    event.transaction.from.toHexString(),
+    event.params.id,
+    rootNode,
+    event.params.expires,
+    TOKEN_ADDRESS_ENS,
+  );
+}
+
+/**
+ * @dev this function maps the NameRegistered from the EthRegistrarControllerOld contract 
+ * @param event ControllerNameRegisteredEventOld from EthRegistrarControllerOld contract
+ */
+export function handleNameRegisteredByControllerOld(event: ControllerNameRegisteredEventOld): void {
+  log.info("handleNameRegisteredByControllerOld: name {} label {} cost {} txhash {}", [event.params.name, event.params.label.toHexString(), event.params.cost.toString(), event.transaction.hash.toHexString()]);
+  let paymentToken: string | null = null
+  if (event.params.cost > BIG_INT_ZERO) {
+    paymentToken = ZERO_ADDRESS;
+  }
+  airstack.domain.trackSetNamePreImage(
+    event.params.name,
+    event.params.label,
+    event.params.cost,
+    paymentToken,
+    event.block.number,
+    event.block.hash.toHexString(),
+    event.block.timestamp,
+    ETHEREUM_MAINNET_ID,
+    rootNode,
+    TOKEN_ADDRESS_ENS,
+    event.transaction.hash.toHexString(),
+    null,
+    null,
+    true,
+  )
+}
+
+/**
+ * @dev this function maps the NameRegistered event from the EthRegistrarController contract
+ * @param event ControllerNameRegisteredEvent from EthRegistrarController contract
+ */
+export function handleNameRegisteredByController(event: ControllerNameRegisteredEvent): void {
```

Comment:
that is because owner changes twice in while registering a domain, once its from 0x to contract and then from contract to actual owner. That is why this is being handled in the domain owner changed handler.